### PR TITLE
Feature: modify app version's configure file separately

### PR DESCRIPTION
**Pull request by pengcong06**
modify app version's configure file separately

### CODE REVIEW

File: pkg/service/app/handler.go [hunk 1]
```diff
@@ -11,6 +11,9 @@ import (
 	"strings"
 	"time"
 
+	"openpitrix.io/openpitrix/pkg/devkit"
+	"openpitrix.io/openpitrix/pkg/devkit/opapp"
+
```
Comment (chilianyi) [inline pkg/service/app/handler.go]: Remove this empty line.

File: pkg/service/app/handler.go [hunk 2]
```diff
@@ -651,16 +654,38 @@ func (p *Server) ModifyAppVersion(ctx context.Context, req *pb.ModifyAppVersionR
 		return nil, err
 	}
 
-	pkg := req.GetPackage().GetValue()
-	if len(pkg) > 0 {
-		_, err = repoiface.LoadPackage(ctx, version.Type, pkg)
+	var mutex bool
```
Comment (chilianyi) [inline pkg/service/app/handler.go]: Is this used for lock?

File: pkg/service/app/handler.go [hunk 3]
```diff
@@ -651,16 +654,38 @@ func (p *Server) ModifyAppVersion(ctx context.Context, req *pb.ModifyAppVersionR
...
+	pkgFiles := req.GetPackageFiles()
+	if len(pkgFiles) > 0 {
```
Comment (zheng1) [inline pkg/service/app/handler.go]: if this app_version is helm type, cannot change with package_files

File: pkg/service/app/handler.go [hunk 4]
```diff
@@ -651,16 +654,38 @@ func (p *Server) ModifyAppVersion(ctx context.Context, req *pb.ModifyAppVersionR
...
-			return nil, gerr.NewWithDetail(ctx, gerr.InvalidArgument, err, gerr.ErrorPackageParseFailed)
+			return nil, err
+		}
+
+		c := &opapp.OpApp{}
+		for name, file := range pkgFiles {
+			content[name] = file
+			if name == devkit.ClusterJsonTmpl {
+				c.ClusterConfTemplate = &opapp.ClusterConfTemplate{Raw: string(file)}
+				if err != nil {
```
Comment (zheng1) [inline pkg/service/app/handler.go]: this check is useless

File: pkg/service/app/handler.go [hunk 5]
```diff
@@ -651,16 +654,38 @@ func (p *Server) ModifyAppVersion(ctx context.Context, req *pb.ModifyAppVersionR
...
+				c.ConfigTemplate = m
+			}
 		}
+		config := c.ConfigTemplate.GetDefaultConfig()
+		err = opapp.ValidateClusterConfTmpl(c.ClusterConfTemplate, config)
 
-		attachmentContent, err := archiveutil.Load(bytes.NewReader(pkg))
 		if err != nil {
-			return nil, gerr.NewWithDetail(ctx, gerr.InvalidArgument, err, gerr.ErrorPackageParseFailed)
+			return nil, gerr.NewWithDetail(ctx, gerr.Internal, err, gerr.ErrorInternalError)
```
Comment (zheng1) [inline pkg/service/app/handler.go]: this error should be "validate package files failed"

File: pkg/service/app/handler.go [hunk 6]
```diff
@@ -651,16 +654,38 @@ func (p *Server) ModifyAppVersion(ctx context.Context, req *pb.ModifyAppVersionR
...
+			if name == devkit.ClusterJsonTmpl {
+				c.ClusterConfTemplate = &opapp.ClusterConfTemplate{Raw: string(file)}
+				if err != nil {
+					return nil, gerr.NewWithDetail(ctx, gerr.Internal, err, gerr.ErrorInternalError)
+				}
+			}
+			if name == devkit.ConfigJson {
+				m, err := opapp.DecodeConfigJson(file)
+				if err != nil {
+					return nil, gerr.NewWithDetail(ctx, gerr.Internal, err, gerr.ErrorInternalError)
```
Comment (zheng1) [inline pkg/service/app/handler.go]: this error should be "decode config.json failed"

File: pkg/service/app/handler.go [hunk 7]
```diff
@@ -670,13 +695,41 @@ func (p *Server) ModifyAppVersion(ctx context.Context, req *pb.ModifyAppVersionR
 
 		_, err = attachmentManagerClient.ReplaceAttachment(ctx, &pb.ReplaceAttachmentRequest{
 			AttachmentId:      version.PackageName,
-			AttachmentContent: attachmentContent,
+			AttachmentContent: content,
```
Comment (zheng1) [inline pkg/service/app/handler.go]: replace attachment with package_files

File: test/app/app_test.go [hunk 8]
```diff
@@ -177,6 +222,16 @@ func testVersionPackage(t *testing.T, appId string) {
...
 
+	modifyAppVersionParams = app_manager.NewModifyAppVersionParams()
+	modifyAppVersionParams.WithBody(
+		&models.OpenpitrixModifyAppVersionRequest{
+			VersionID:    versionId2,
+			PackageFiles: preparePackageFile(t, "0.0.4"),
+		})
+	_, err = client.AppManager.ModifyAppVersion(modifyAppVersionParams, nil)
+
+	testutil.NoError(t, err, Service)
```
Comment (zheng1) [inline test/app/app_test.go]: need to check the failed case

File: pkg/service/app/handler.go (hunk 7 above)
Comment (pengcong06) [reply pkg/service/app/handler.go]: what means?
three steps to replace attachment with package_files 
1.use proxy to get whole package files(m1 = map[string][]byte);
2.use m2 = map[string][]byte passed in to replace the old one, eg.m1["config.json"]=m2["config.json"];
3.save m1

Comment (pengcong06) [reply pkg/service/app/handler.go]: > replace attachment with package_files

what means?
three steps to replace attachment with package_files
1.use proxy to get whole package files(m1 = map[string][]byte);
2.use m2 = map[string][]byte passed in to replace the old one, eg.m1["config.json"]=m2["config.json"];
3.save m1

or,you means that should add a rpc named ReplaceAttachmentWithPackageFiles to handle?

**Pull request closed without merging**